.github/workflows/pytest_multi_gateway.yml

```diff
@@ -0,0 +1,59 @@
+name: Run pytest_multi_gateway
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  pytest_multi_gateway:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          # git submodule update --init --recursive
+          submodules: recursive
+
+      - name: Build images
+        run: |
+          make build
+
+      - name: Setup huge pages
+        run: |
+          make setup
+
+      - name: Start ceph-cluster
+        run: |
+          docker-compose up --detach ceph-cluster
+
+      - name: Wait for the ceph-cluster container to become healthy
+        run: |
+          while true; do
+            container_status=$(docker inspect --format='{{.State.Health.Status}}' ceph-cluster)
+            if [[ $container_status == "healthy" ]]; then
+            break
+          else
+          # Wait for a specific time before checking again
+            sleep 1
+          fi
+          done
```

Comment:
For pytest/unit-test specific testing, we could probably skip the building of the containers and just run the test in a lightweight python-enabled container. The idea of this would be to provide extremely fast feedback (<1 min after pushing the PR), while anything integration-related (spdk-based) would take minutes to complete.

BTW, I see different actions for each test. Couldn't we use [a matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) for that instead?

Reply:
@epuertat,  matrix is a kewl idea 😀, looks good  [here](https://github.com/baum/ceph-nvmeof/commit/ca0e9a0802ca4ab36dec61918b434ed359d8f78c), thanks